src/axom/core/tests/core_execution_space.cpp

```diff
@@ -0,0 +1,309 @@
+template < typename ExecSpace,
+           typename LoopPolicy,
+           typename Loop2dPolicy,
+           typename Loop3dPolicy,
+           typename ReducePolicy,
+           typename AtomicPolicy,
+           typename SyncPolicy >
+void check_execution_mappings( int expectedAllocatorID, bool is_async )
+{
+  std::cout << "checking execution space: " <<
+               axom::execution_space< ExecSpace >::name() << std::endl;
+
+  using loop_pol   = typename axom::execution_space< ExecSpace >::loop_policy;
+  using loop2d_pol = typename axom::execution_space< ExecSpace >::loop2d_policy;
+  using loop3d_pol = typename axom::execution_space< ExecSpace >::loop3d_policy;
+  using reduce_pol = typename axom::execution_space< ExecSpace >::reduce_policy;
+  using atomic_pol = typename axom::execution_space< ExecSpace >::atomic_policy;
+  using sync_pol   = typename axom::execution_space< ExecSpace >::sync_policy;
+
+  bool valid_loop_policy   = std::is_same< loop_pol, LoopPolicy >::value;
+  bool valid_loop2d_policy = std::is_same< loop2d_pol, Loop2dPolicy >::value;
+  bool valid_loop3d_policy = std::is_same< loop3d_pol, Loop3dPolicy >::value;
+  bool valid_reduce_policy = std::is_same< reduce_pol, ReducePolicy >::value;
+  bool valid_atomic_policy = std::is_same< atomic_pol, AtomicPolicy >::value;
+  bool valid_sync_policy   = std::is_same< sync_pol, SyncPolicy >::value;
+
+  EXPECT_TRUE( valid_loop_policy );
+  EXPECT_TRUE( valid_loop2d_policy );
+  EXPECT_TRUE( valid_loop3d_policy );
+  EXPECT_TRUE( valid_reduce_policy );
+  EXPECT_TRUE( valid_atomic_policy );
+  EXPECT_TRUE( valid_sync_policy );
+
+  const bool async = axom::execution_space< ExecSpace >::async();
+  EXPECT_EQ( async, is_async );
+
+  int allocatorID = axom::execution_space< ExecSpace >::allocatorID();
+  EXPECT_EQ( expectedAllocatorID, allocatorID );
+}
+
+} /* end anonymous namespace */
+
+//------------------------------------------------------------------------------
+//  UNIT TESTS
+//------------------------------------------------------------------------------
+
+//------------------------------------------------------------------------------
+TEST( core_execution_space, check_valid )
+{
+  check_valid< axom::SEQ_EXEC >( );
+
+#if defined(AXOM_USE_OPENMP) && defined(AXOM_USE_RAJA)
+  check_valid< axom::OMP_EXEC >( );
+#endif
+
+#if defined(AXOM_USE_CUDA) && defined(AXOM_USE_RAJA)
+  check_valid< axom::CUDA_EXEC< 256 > >( );
+  check_valid< axom::CUDA_EXEC< 256, axom::ASYNC > >( );
+#endif
+}
+
+//------------------------------------------------------------------------------
+TEST( core_execution_space, check_invalid )
+{
+  struct InvalidSpace { };
+  check_invalid< InvalidSpace >( );
+}
+
+//------------------------------------------------------------------------------
+#ifdef AXOM_USE_RAJA
+TEST( core_execution_space, check_seq_exec )
+{
+  check_valid< axom::SEQ_EXEC >( );
+
+  constexpr bool IS_ASYNC = false;
+
+  /* *INDENT-OFF* */
+  using raja_loop2d_policy =
```

Comment:
Why use RAJA policies here and not the axom ones you defined?

Reply:
This code tests the mapping of the `execution_space` traits class to the corresponding RAJA policies. So, the expected RAJA policies are hard-coded in this test. Makes sense?